src/compat/reversediff.jl

```diff
@@ -208,9 +227,20 @@ end
     return y, (wrap_chainrules_output ∘ Base.tail ∘ dy)
 end
 
+@grad_from_chainrules update_triu_from_vec(vals::TrackedVector{<:Real}, k::Int, dim::Int)
+
+@grad_from_chainrules _link_chol_lkj(x::TrackedMatrix)
+@grad_from_chainrules _inv_link_chol_lkj(x::TrackedVector)
+
 # NOTE: Probably doesn't work in complete generality.
 wrap_chainrules_output(x) = x
 wrap_chainrules_output(x::ChainRulesCore.AbstractZero) = nothing
 wrap_chainrules_output(x::Tuple) = map(wrap_chainrules_output, x)
 
+# HACK: To make it work for julia v1.6 .
+# This dispatch does not wrap X in Hermitian before calling cholesky. 
+# cholesky does not work with AbstractMatrix in julia v1.6,
+# and it would error with Hermitian{ReverseDiff.TrackedArray}.
+cholesky_factor(X::ReverseDiff.TrackedArray) = cholesky_factor(cholesky(X))
```

Comment:
I'd suggest figuring out the Julia version from which on the linked PR was available (see, e.g., Compat.jl for instructions) and then wrap this fix in a `if VERSION < v"..."` block. This would unify the AD backends on more recent versions at least.